### PR TITLE
Add meal button in nav bar

### PR DESCRIPTION
**Pull request by Gmiller290488**
Added a plus button in the meal list and map view which opens the add Meal screen as a sheet.

I went with displaying the AddMeal in a sheet because it felt weird to push it... it would need a back arrow and you could change tabs but still be on the AddMeal screen when you switched tabs back again and I think that this plus button here is only a temporary situation anyway :-)

Also created the package for the Settings, refactored some duplicated code, and removed some unused code

small-addmeal

### CODE REVIEW

File: iosHackthonApp/Modules/Features/MealList/Sources/MealList/MealListView.swift [hunk 1]
```diff
@@ -26,13 +27,22 @@ public struct MealListView: View {
...
+                                    }) {
+                                        Image(systemName: Strings.Common.Images.add)
+                                    },
+                                trailing:
+                                    Button(action: {
+                                        self.viewModel.loadMeals()
+                                    }, label: {
+                                        Image(systemName: Strings.MealListScreen.Images.reload)
+                                    }))
+                .foregroundColor(.white)
```
Comment (Gmiller290488) [inline iosHackthonApp/Modules/Features/MealList/Sources/MealList/MealListView.swift]: will fix this indentation - just want to see if CI passes 😬

File: iosHackthonApp/Modules/Features/MealList/Sources/MealList/MealListViewModel.swift [hunk 2]
```diff
@@ -40,7 +41,7 @@ public final class MealListViewModel: ObservableObject {
             guard
                 let meals = meals,
                 error == nil else {
-                assertionFailure(error!.localizedDescription)
+                print(error!.localizedDescription)
```
Comment (KaneCheshire) [inline iosHackthonApp/Modules/Features/MealList/Sources/MealList/MealListViewModel.swift:46]: Hmm does this mean we should maybe be showing some sort of error state to the user?

**Review by KaneCheshire — APPROVED** [review]

File: iosHackthonApp/Modules/Features/MealList/Sources/MealList/MealListViewModel.swift (hunk 2 above)
Comment (Gmiller290488) [reply iosHackthonApp/Modules/Features/MealList/Sources/MealList/MealListViewModel.swift:46]: Yeah, we should show an alert or something. I added this just as a catch to remind me that I didn't start the `json-server` yet 😬 but I think that assertion was causing my CI problems last night/this morning which is why I changed it to a print statement

File: iosHackthonApp/Modules/Features/AddMeal/Sources/AddMeal/AddMealView.swift [hunk 3]
```diff
@@ -46,46 +46,48 @@ public struct AddMealView: View {
...
-                    AddressTextFieldGroup(locationManager: locationManager, address: $address, latitude: $latitude, longitude: $longitude)
-                    Spacer()
-                    GeometryReader { geometry in
-                        Button(action: {
-                            postMeal(meal: createMeal())
-                        }) {
-                            Text(Strings.AddMealScreen.addMeal)
-                                .modifier(AddButtonStyle(width: geometry.size.width,
-                                                         backgroundColor: buttonColor))
+        NavigationView {
```
Comment (Gmiller290488) [inline iosHackthonApp/Modules/Features/AddMeal/Sources/AddMeal/AddMealView.swift:49]: The only change in here is that I added the NavigationView

**Review by raynelle03 — APPROVED** [review]

**Pull request merged**